recipes/zxing-cpp/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ZXingCppConan(ConanFile):
+    name = "zxing-cpp"
+    homepage = "https://github.com/nu-book/zxing-cpp"
+    description = "c++14 port of ZXing, a barcode scanning library"
+    topics = ("conan", "zxing", "barcode", "scanner", "generator")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_encoders": [True, False],
+        "enable_decoders": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enable_encoders": True,
+        "enable_decoders": True,
+    }
+    generators = "cmake"
+
+    _cmake = None
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _compiler_cpp14_support = {
+        "gcc": "5",
+        "Visual Studio": "14",
+        "clang": "3.4",
+        "apple-clang": "3.4",
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if str(self.settings.compiler.cppstd) in ("98", "gnu98", "11", "gnu11"):
+            raise ConanInvalidConfiguration("This library requires at least c++ 14. The requested c++ standard is too low.")
+        try:
+            minimum_required_version = self._compiler_cpp14_support[str(self.settings.compiler)]
+            if self.settings.compiler.version < tools.Version(minimum_required_version):
+                raise ConanInvalidConfiguration("This compiler is too old. This library needs a compiler with c++14 support")
+        except KeyError:
```

Comment:
The author project doesn't support or it's because we don't have the cppstd version listed? I prefer raising a warning only, like in other recipes. Otherwise, we will need to update this recipe for each new compiler.

Reply:
These older compilers do not support c++14.

Reply:
Check my suggestions. Would that be enough?

Reply:
Have I addressed your concern in these latest commits?